src/main/java/org/traccar/protocol/TranSyncProtocolDecoder.java

```diff
@@ -0,0 +1,218 @@
+        position.set(Position.PREFIX_IN + 3, BitUtil.check(zeroByte, 2));
+        if (BitUtil.check(zeroByte, 3)) position.set(Position.ALARM_POWER_OFF, true);
+        position.set(Position.KEY_IGNITION, BitUtil.check(zeroByte, 4));
+        position.set("gpsFix", BitUtil.check(zeroByte, 7));
+        int oneByte =  buf.readUnsignedByte();
+        int deviceAlert =  buf.readUnsignedByte();
+        if (deviceAlert > 0) { setDeviceAlert(position, deviceAlert); }
+        int threeByte =  buf.readUnsignedByte();
+        if (!((threeByte & 0b10000000) == 0)) position.set("restored", true);
+        if ((threeByte & 0b00100000) != 0) {
+            position.set(Position.ALARM_GPS_ANTENNA_CUT, true);
+            position.set("gpsAlert", true);
+        }
+        position.set("model", getTrackerModel(threeByte & 0b00001111));
+        int gsmSignalStrength = buf.readUnsignedByte();
+        position.set(Position.KEY_BATTERY, (double) (buf.readUnsignedByte() / 10));
+        position.set(Position.KEY_SATELLITES, buf.readUnsignedByte());
+        position.set(Position.KEY_HDOP, buf.readUnsignedByte());
+        position.set(Position.PREFIX_ADC + 1, (short) buf.readUnsignedShort());
+        boolean isOptionalParameters = (buf.readableBytes() > 2);
+        setNetwork(position, lacInt, cellIdInt, mobileNetworkCode, gsmSignalStrength);
+        if (isOptionalParameters) { //  Always True
+            int odometerIndex = buf.readUnsignedByte();
+            int odometerLength = buf.readUnsignedByte();
+            if (odometerLength > 0) {
+                String odometerReading = ByteBufUtil.hexDump(buf, buf.readerIndex(), odometerLength).toUpperCase().trim().replaceFirst("^0+(?!$)", "");
+                int odometer = Integer.parseInt(odometerReading);
+                buf.readBytes(odometerLength);
+                position.set(Position.KEY_ODOMETER, odometer);
+            }
+            if ((buf.readableBytes() > 2)) {
+                int rfidIndex = buf.readUnsignedByte();
+                int rfidLength = buf.readUnsignedByte();
+                if(rfidLength > 0) {
+                    String rfidTagName = ByteBufUtil.hexDump(buf, buf.readerIndex(), rfidLength).toUpperCase().trim();
+                    buf.readBytes(rfidLength);
+                    position.set("tag", rfidTagName);
+                }
+            }
+            if ((buf.readableBytes() > 5)) {
+                int adcTwoIndex =  buf.readUnsignedByte();
+                int adcTwoLength =  buf.readUnsignedByte();
+                if (adcTwoLength > 0){
+                    position.set(Position.PREFIX_ADC + 2, buf.readUnsignedShort());
+                }
+            }
+        }
+
+        return position;
+    }
+
+    @Override
+    protected Object decode( Channel channel, SocketAddress remoteAddress, Object msg) throws Exception {
+        ByteBuf buffer = (ByteBuf) msg;
+        return parseData(channel, remoteAddress, buffer);
```

Comment:
You can probably do parsing directly in this method. Doesn't make sense to just call `parseData` in here.